packages/svelte-query/src/useQueryClient.ts

```diff
@@ -0,0 +1,14 @@
+import type { QueryClient } from '@tanstack/query-core'
+import { client } from './store'
+
+export function useQueryClient(): QueryClient {
+  let queryClient!: QueryClient
+  client.subscribe((val) => {
```

Comment:
Thanks for the feedback. My initial approach was using a svelte component but I can't get the build system to package .svelte files. I really like using sveltekit's package script. It makes packaging svelte libraries seamless.

Reply:
I think the best approach is indeed using a svelte component.

Another approach, maybe easier, could be to use `onMount` and `onDestroy` hooks.
However this means that the `useQueryClient` will need to be called inside a Svelte component.

Reply:
Finally, maybe the best solution would be to use `getContext` and `setContext`.

In this case:
1. we export a method `initQueryClient(options)` which allows to create a `QueryClient` with custom options and call `setContext` to provide the `QueryClient` to child components
2. we change `useQueryClient()` to use `getContext` to retrieve the current `QueryClient`

This approach has multiple advantages:
- allow to provide custom options
- allow to have multiple `QueryClient` with different options
- avoid having a static global `QueryClient`
- avoid creating "ghost" subscriptions

Reply:
On "ghost subscriptions", I've been trying this PR out in one of our projects and found one design decision that we might need to rethink, unless your suggestion solves this as well:

Here's the scenario:

`let myQuery = useQuery({enabled: false, queryFn: => (....)})` — let's say queryFn depends on some $reactive variables, so when `useQuery` is called all of them are undefined, so we disable it by default or else the query would fail the first time. We actually want the query to run each time one of those vars changes as they are probably bound to a user input.

Unlike in a React component, `useQuery` will only ever be called once when the component is initialized. We could assigned it reactively e.g. `$: myQuery = useQuery(...)`, which would wait until the dependent variables are initialized and it would appear to work, however, the observer and subscriptions would be recreated each time leading to a memory leak.

One workaround would be to change the API so that the query is initialized separately from actually being called:

```ts
let myQuery = useQuery({queryFn: ({variables})=> {...}})
$: myQuery.query({variables: {a: $aVar}})
```

This way the `queryFn` can receive the inputs it needs when they change, triggering a key change and query. 

Another idea: make `queryFn` a store, so any reactive change would trigger a query.